llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -130,6 +130,7 @@ struct ImagePropSaveInfo {
   bool SetSpecConstAtRT;
   bool SpecConstsMet;
   bool EmitKernelParamInfo;
+  bool IsEsimdKernel;
```

Comment:
general comment: there will be no way to tell the post link not to perform SYCL/ESIMD splitting. It might not be always necessary (and ideally, GPU BEs will accept both kinds of kernels in one module in future), so I'd suggest to add something like `-no_sycl_esimd_split` to preserve previous behavior.

Reply:
I introduced a new `-split-esimd` option that controls SYCL-ESIMD splitting.